src/dodal/beamlines/i18.py

```diff
@@ -0,0 +1,218 @@
+set_directory_provider(
+    StaticVisitDirectoryProvider(
+        BL,
+        Path("/dls/i18/data/2024/cm37264-2/bluesky"),
+        client=DirectoryServiceClient("http://i18-control:8088/api"),
+    )
+)
+
+
+def synchrotron(
+    wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
+) -> Synchrotron:
+    return device_instantiation(
+        Synchrotron,
+        "synchrotron",
+        "",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+    )
+
+
+def undulator(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> Undulator:
+    return device_instantiation(
+        Undulator,
+        "undulator",
+        f"{BeamlinePrefix(BL).insertion_prefix}-MO-SERVC-01:",
+        # add CURRGAPD
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        bl_prefix=False,
+        poles=80,
+        length=2.0,
+    )
+
+
+@skip_device()
+def slits_1(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> Slits:
+    return numbered_slits(
+        1,
+        # BL18I-AL-SLITS-01
+        wait_for_connection,
+        fake_with_ophyd_sim,
+    )
+
+
+# Must document what PandAs are physically connected to
+# See: https://github.com/bluesky/ophyd-async/issues/284
+@skip_device()
+def panda1(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> HDFPanda:
+    return device_instantiation(
+        HDFPanda,
+        "panda1",
+        "-MO-PANDA-01:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        directory_provider=get_directory_provider(),
+    )
+
+
+@skip_device()
+def xspress3(
+    wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
+) -> Xspress3:
+    """
+    16 channels Xspress3 detector
+        -EA-XPS-02:CAM:MaxSizeX_RBV
+      also ArraySize
+      also :CONNECTED
+    """
+
+    return device_instantiation(
+        Xspress3,
+        # prefix="-EA-DET-03:",
+        prefix="-EA-XPS-02:",
+        name="Xspress3",
+        num_channels=16,
+        wait=wait_for_connection,
+        fake=fake_with_ophyd_sim,
+    )
+
+
+def dcm(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> DoubleCrystalMonochromator:
+    return device_instantiation(
+        DoubleCrystalMonochromator,
+        "dcm",
+        "",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        bl_prefix=False,
+        motion_prefix=f"{BeamlinePrefix(BL).beamline_prefix}-MO-DCM-01:",
+        temperature_prefix=f"{BeamlinePrefix(BL).beamline_prefix}-DI-DCM-01:",
+        crystal_1_metadata=CrystalMetadata(
+            usage="Bragg",
+            type="silicon",
+            reflection=(1, 1, 1),
+            d_spacing=(3.13475, "nm"),
+        ),
+        crystal_2_metadata=CrystalMetadata(
+            usage="Bragg",
+            type="silicon",
+            reflection=(1, 1, 1),
+            d_spacing=(3.13475, "nm"),
+        ),
```

Comment:
Is it correct that the DCM on i18 has the same reflections and `d_spacing` as i22? Probably a question for the i18 scientists... If it is true then maybe these are wider defaults we should put in the DCM device?

Reply:
covered there
https://github.com/DiamondLightSource/dodal/pull/722#discussion_r1700003058